src/Stratis.Bitcoin.Features.Apps/AppsStore.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Features.Apps.Interfaces;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Bitcoin.Utilities.Extensions;
+
+namespace Stratis.Bitcoin.Features.Apps
+{
+    /// <summary>
+    /// Responsible for storing StratisApps as read from the current running Stratis folder.
+    /// </summary>
+    public class AppsStore : IAppsStore
+    {
+        private readonly ILogger logger;
+        private List<IStratisApp> applications;
+        private readonly DataFolder dataFolder;
+        private const string ConfigFileName = "stratisApp.json";
+
+        public AppsStore(ILoggerFactory loggerFactory, DataFolder dataFolder)
+        {
+            this.dataFolder = dataFolder;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);            
+        }
+
+        public IEnumerable<IStratisApp> Applications
+        {
+            get
+            {
+                if (this.applications == null)
+                    this.Load();
+
+                return this.applications;
+            }
+        }
+
+        private void Load()
+        {
+            try
+            {
+                Debug.Assert(this.applications==null);
```

Comment:
Instead of throwing an exception here, can you return the applications instead?

Reply:
Not quite, but have made cleaner. 

Just pushed.